omc3/correction/response_madx.py

```diff
@@ -106,7 +108,19 @@ def _generate_madx_jobs(
             current_job += f"{var} = {var}{-delta_k:+.15e};\n\n"
 
         if proc_idx == num_proc - 1:
-            current_job += f"twiss, file='{str(temp_dir / 'twiss.0')}';\n"
+            if dodeltap and abs(accel_inst.dpp) > 0: # This is here only for multiple iteration of the global correction
+                # By including dpp here, it means that if deltap is in variables and dpp is not 0, the reference (twiss.0) changes 
+                if len(variables) > 0: # To ensure someone is not calculating response matrix of knobs with a twiss.0 that includes dpp
+                    raise NotImplementedError(f"{DELTAP_NAME} and dpp != 0, with additional variables is not implemented.")
+                current_job += accel_inst.get_update_deltap_script() 
+            current_job += f"twiss, file='{str(temp_dir / 'twiss.0')}', deltap={DELTAP_NAME};\n"
+            
+            if dodeltap: # If DELTAP_NAME is in variables, we run this in the last iteration
+                # Due to the match and correction of the orbit, this needs to be run at the end of the process
+                incr_dict[DELTAP_NAME] = delta_k
+                current_job += f"{DELTAP_NAME} = {delta_k:.15e};\n" # Set deltap to delta_k
+                current_job += accel_inst.get_update_deltap_script() # This will add accel_inst.dpp to deltap and do twiss, correct, match
+                current_job += f"twiss, deltap={DELTAP_NAME}, file='{str(temp_dir/f'twiss.{DELTAP_NAME}')}';\n"
```

Comment:
So I see it as these two scenarios:
- There is a dpp set in the model, so we have to make sure the reference twiss file also has the correct settings with the orbit correctors and tune. Then we do a `+delta_k` from this dpp.
- There is no dpp set, so we have a normal reference twiss, and just do a twiss with a `+delta_k` from a 0 dpp.

I believe that this function runs a self contained madx instance, so, since all this happens in the final part of the final process should mean that there is no leakage of changes to this dpp variable. Therefore, yes, this must happen 

The lack of possibility of other variables is because I am changing the reference twiss. To ensure consistency, I would have to also adjust the orbit and include dpp when using the other variables. This is possible, but then you come across the problem that the orbit corrector strengths have changed, so might want to reset them, and then I would want to do a reset of dpp to be sure.

In theory, it could be better to include this 
```python
if dodeltap and abs(accel_inst.dpp) > 0: # This is here only for multiple iteration of the global correction
```
before the for loop (not forgetting to include dpp in the twiss for the variables). This means that the whole MAD-X run would be run with the adjustments to orbit and dpp.